platform/linux-generic/pktio/dpdk.c

```diff
@@ -549,7 +553,7 @@ static inline int mbuf_to_pkt(pktio_entry_t *pktio_entry,
 		if (pktio_cls_enabled(pktio_entry))
 			copy_packet_cls_metadata(&parsed_hdr, pkt_hdr);
 		else if (parse_layer != ODP_PROTO_LAYER_NONE)
-			packet_parse_layer(pkt_hdr, parse_layer);
+			dpdk_packet_parse_layer(pkt_hdr, mbuf, parse_layer);
```

Comment:
I have been thinking about similar change in generic code. However in generic code some PktIOs call parsing function using buffer, which is not a part of the packet (yet). So it is not possible to always pass odp_packet_t/odp_buffer_t to parsing functions.